ckan/logic/__init__.py

```diff
@@ -1000,3 +1006,50 @@ def index_remove_package(id_: str):
     from ckan.lib import search
     index = search.index_for('Package')
     index.remove_dict({'id': id_})
+
+
+class ContextCache:
+    """Cache for storing and retrieving values in the context."""
```

Comment:
`fresh_context` was written to avoid this pattern. We shouldn't be using the context dict as a way to pass values up and down the call stack because that leads to bugs that are very difficult to understand and reproduce.

What's the specific feature that needs to pass values this way and why?